src/plugins/condition/ConditionManager.js

```diff
@@ -202,12 +208,18 @@ export default class ConditionManager extends EventEmitter {
         this.persistConditions();
     }
 
-    removeCondition(index) {
-        let condition = this.conditions[index];
-        condition.destroy();
-        this.conditions.splice(index, 1);
-        this.conditionSetDomainObject.configuration.conditionCollection.splice(index, 1);
-        this.persistConditions();
+    removeCondition(id) {
+        let index = this.conditions.findIndex(item => item.id === id);
+        if (index > -1) {
+            this.conditions[index].destroy();
+            this.conditions.splice(index, 1);
+        }
+
+        index = this.conditionSetDomainObject.configuration.conditionCollection.findIndex(item => item.id === id);
```

Comment:
I'd use a separate variable definition for condition index and conditionCollection index.

Reply:
Done.